app/src/main/java/ch/epfl/cs311/wanderwave/model/remote/ProfileConnection.kt

```diff
@@ -26,14 +26,11 @@ class ProfileConnection(
   override val db = database
   private val coroutineScope = CoroutineScope(Dispatchers.IO + SupervisorJob())
 
-  override fun isUidExisting(spotifyUid: String, callback: (Boolean, Profile?) -> Unit) {
+  override fun isUidExisting(firebaseUid: String, callback: (Boolean, Profile?) -> Unit) {
     db.collection("users")
-        .whereEqualTo("spotifyUid", spotifyUid)
+        .document(firebaseUid)
         .get()
-        .addOnSuccessListener { documents ->
-          val isExisting = documents.size() > 0
-          callback(isExisting, if (isExisting) documentToItem(documents.documents[0]) else null)
-        }
+        .addOnSuccessListener { document -> callback(true, documentToItem(document)) }

```

Comment:
Like we sais we should change this in the future, I don't know if you want to do it now, in your branch or for a future issue.

Reply:
I'll wait for the error handling in the FirebaseConnection, that way we don't need this function in the first place.